src/Database/Type/BoolType.php

```diff
@@ -67,7 +67,10 @@ public function toDatabase($value, Driver $driver)
             return (bool)$value;
         }
 
-        throw new InvalidArgumentException('Cannot convert value to bool');
+        throw new InvalidArgumentException(sprintf(
+            'Cannot convert value of type `%s` to bool',
+            getTypeName($value)
```

Comment:
Wouldnt it then also make sense to include the value itself directly? Or is that a bit too much then as debug info?

Reply:
I wanted to do that, but how could we archive this? It's just the base \InvalidArgumentException and doesn't really have a way to attach arbitrary data to it? Also does the error renderer has the capability of showing that value? Especially in this case of wrong type values, know the content of `$value` would make debugging the cause extremely helpful.

Reply:
Yeah, maybe a bit out of scope - or even overkill - to use some Debugger::exportVar().
Hopefully the stracktrace contains the caller arguments.